src/compiler/objective_c_generator.cc

```diff
@@ -176,65 +176,71 @@ void PrintMethodImplementations(Printer *printer,
 
 string GetHeader(const ServiceDescriptor *service, const string prefix) {
   string output;
-  grpc::protobuf::io::StringOutputStream output_stream(&output);
-  Printer printer(&output_stream, '$');
+  {
+    // Scope the output stream so it closes and finalizes output to the string.
```

Comment:
Note that this isn't a grpc issue per-se. I'm 94.3% certain we could file that as a compiler bug. There's no way moving the string before calling the local destructors is an acceptable behavior. The bad part we're exhibiting here (and I just realized that) is that we're taking addresses of temporaries, which is supposed to be a compiler warning.

If you want to be more specific in that comment, I'd rather suggest saying this is to solve a Visual Studio problem with the destruction ordering of temporaries vs the move operation.